### PR TITLE
VOLKized nlog10

### PR DESCRIPTION
**Pull request by marcusmueller**
The inner VOLK demanded improvement of this.

Still not perfect; a volk_32f_s32f_log2_scaled kernel would be better.

### CODE REVIEW

File: gr-blocks/lib/nlog10_ff_impl.cc [hunk 1]
```diff
@@ -39,24 +40,40 @@ namespace gr {
...
+    {
+      d_prefactor = n / log2f(10.0f);
     }
 
     int
     nlog10_ff_impl::work(int noutput_items,
 			      gr_vector_const_void_star &input_items,
 			      gr_vector_void_star &output_items)
     {
+      #define novolk  0
```
Comment (jmcorgan) [inline gr-blocks/lib/nlog10_ff_impl.cc]: This looks like a stray debug line that should have been removed.

File: gr-blocks/lib/nlog10_ff_impl.cc [hunk 2]
```diff
@@ -39,24 +40,40 @@ namespace gr {
...
-      float n = d_n;
-      float k = d_k;
-
-      for (int i = 0; i < noi; i++)
-	out[i] = n * log10(std::max(in[i], (float) 1e-18)) + k;
 
+      volk_32f_log2_32f(out, in, noi);
+      volk_32f_s32f_multiply_32f(out, out, d_prefactor, noi);
+      for(int i = 0; i < noi; ++i) {
+        out[i] += d_k;
```
Comment (jmcorgan) [inline gr-blocks/lib/nlog10_ff_impl.cc]: Is there not a vectorized add_const kernel?

**Comment by awalls-cx18** [comment]
If d_k == 0, no need to loop through adding, so skip the loop in that case. :)

File: gr-blocks/lib/nlog10_ff_impl.cc (hunk 2 above)
Comment (marcusmueller) [inline gr-blocks/lib/nlog10_ff_impl.cc]: no.

File: gr-blocks/lib/nlog10_ff_impl.cc (hunk 1 above)
Comment (marcusmueller) [inline gr-blocks/lib/nlog10_ff_impl.cc]: true. Will die.

**Comment by jmcorgan** [comment]
Also, this seems like it could go on master instead of next.

**Comment by marcusmueller** [comment]
> Also, this seems like it could go on master instead of next.

The original test case asserts `10log10(-10) == -180`. My code of course does `10log10(-10) == nan`. I do think I'm right, but I'd like your input on this, since it breaks compat.

**Comment by jmcorgan** [comment]
Good point.  I agree the correct behavior is NaN, but consider it likely users aren't checking for this possibility, including ourselves.

**Pull request merged**